Userland/Libraries/LibCore/System.h

```diff
@@ -206,6 +210,55 @@ ErrorOr<void> unlockpt(int fildes);
 ErrorOr<void> access(StringView pathname, int mode);
 ErrorOr<DeprecatedString> readlink(StringView pathname);
 
+template<Pointer T>
+struct OwnPtrWithCustomDeleter {
+    AK_MAKE_NONCOPYABLE(OwnPtrWithCustomDeleter);
+
+public:
+    OwnPtrWithCustomDeleter(T ptr, AK::Function<void(T)> deleter)
+        : m_ptr(ptr)
+        , m_deleter(move(deleter))
+    {
+    }
+
+    OwnPtrWithCustomDeleter(OwnPtrWithCustomDeleter&& other)
+    {
+        swap(m_ptr, other.m_ptr);
+        swap(m_deleter, other.m_deleter);
+    };
+
+    ~OwnPtrWithCustomDeleter()
+    {
+        if (m_deleter && m_ptr)
+            m_deleter(m_ptr);
```

Comment:
Maybe `VERIFY()` the deleter? Otherwise it seems like it might be possible for a leak to slip through: 
```suggestion
        if (m_ptr) {
            VERIFY(m_deleter);
            m_deleter(m_ptr);
        }
```